File: app/src/main/java/com/platdmit/simplecloudmanager/data/api/ApiManager.java
package com.platdmit.simplecloudmanager.data.api;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public abstract class ApiManager {
    private static final String TAG = ApiManager.class.getSimpleName();
    private static final String BASE_URL = "https://api.simplecloud.ru/v3/";
    private static String API_KEY;

    private static OkHttpClient getOkHttpInstance() {
        return new OkHttpClient.Builder()
                .addInterceptor(chain -> {
                    Request originalRequest = chain.request();
                    Request.Builder builder = originalRequest.newBuilder().header("Authorization", "Bearer " + API_KEY);
                    Request newRequest = builder.build();
                    return chain.proceed(newRequest);
                }).addInterceptor(getOkHttpLogLevel()).build();
    }

    private static HttpLoggingInterceptor getOkHttpLogLevel(){
        HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
        httpLoggingInterceptor.level(HttpLoggingInterceptor.Level.BASIC);
        return httpLoggingInterceptor;
    }

    private static Retrofit getRetrofitInstance() {
        return new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .client(getOkHttpInstance())
                .addConverterFactory(GsonConverterFactory.create())
                .build();
    }

    @SuppressWarnings("unchecked")
    public static <T> T getApiPoint(Class rest, String api_key){
        API_KEY = api_key;
        return (T) getRetrofitInstance().create(rest);
    }
}

File: data/src/main/java/com/platdmit/data/api/ApiManager.kt
package com.platdmit.data.api

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

@Deprecated("Implementation updated, moved to di")
object ApiManager {
    private val TAG = ApiManager::class.java.simpleName
    private const val BASE_URL = "https://api.simplecloud.ru/v3/"
    private var API_KEY: String? = null
    private val okHttpInstance: OkHttpClient
        get() = OkHttpClient.Builder()
                .addInterceptor{
                    val newRequest = it.request().newBuilder().header("Authorization", "Bearer $API_KEY").build()
                    it.proceed(newRequest)
                }
                .addInterceptor(okHttpLogLevel).build()

    private val okHttpLogLevel: HttpLoggingInterceptor
        get() {
            val httpLoggingInterceptor = HttpLoggingInterceptor()
            httpLoggingInterceptor.level = HttpLoggingInterceptor.Level.BASIC
            return httpLoggingInterceptor
        }

    private val retrofitInstance: Retrofit
        get() = Retrofit.Builder()
                .baseUrl(BASE_URL)
                .client(okHttpInstance)
                .addConverterFactory(GsonConverterFactory.create())
                .build()

    @JvmStatic
    fun <T> getApiPoint(rest: Class<T>, api_key: String): T {
        API_KEY = api_key
        return retrofitInstance.create(rest) as T
    }
}